src/Development/IDE/Core/Compile.hs

```diff
@@ -740,19 +731,20 @@ loadInterface session ms sourceMod objNeeded regen = do
             -- one-shot mode.
             | not (mi_used_th iface) || SourceUnmodifiedAndStable == sourceMod
             -> do
-             linkable <-
-               if objNeeded
-               then liftIO $ findObjectLinkableMaybe (ms_mod ms) (ms_location ms)
-               else pure Nothing
-             let objUpToDate = not objNeeded || case linkable of
+             linkable <- case linkableNeeded of
+               Just ObjectLinkable -> liftIO $ findObjectLinkableMaybe (ms_mod ms) (ms_location ms)
+               _ -> pure Nothing
+
+             -- We don't need to regenerate if the object is up do date, or we don't need one
+             let objUpToDate = isNothing linkableNeeded || case linkable of
                    Nothing -> False
                    Just (LM obj_time _ _) -> obj_time > ms_hs_date ms
```

Comment:
This is where we compare the linkable timestamp to decide if it's up to date

Reply:
We only do that for object code (which is not used anymore), and after we look it up from disk. The time here is the modification time of the file, as looked up by `findObjectLinkableMaybe`. Also, when we generate the object linkable ourselves (instead of looking it up from disk), we use `getModificationTime` to get the time.